SwordImportExportPlugin.inc.php

```diff
@@ -190,19 +179,23 @@ public function display($args, $request) {
 				);
 
 				$errors = array();
-				// select at least one article
-				$articleIds = $request->getUserVar('articleId');
-				if (empty($articleIds)) {
+
+				$submissionIds = $request->getUserVar('selectedSubmissions');
+				// select at least one submission
+				if (empty($submissionIds)) {
 					$errors[] = array(
 						'title' => __('plugins.importexport.sword.requiredFieldErrorTitle'),
 						'message' => __('plugins.importexport.sword.requiredFieldErrorMessage'),
 					);
 				}
 				else {
-					foreach ($articleIds as $articleId) {
-						$publishedArticle = $publishedArticleDao->getByArticleId($articleId);
+					foreach ($submissionIds as $submissionId) {
+						$submission = $submissionDao->getById($submissionId);
+						if ($submission->getJournalId() != $request->getJournal()->getId()) {
```

Comment:
Better to use `$submission->getContextId`. `$submission->getJournalId()` is OJS-specific and is marked deprecated as of 3.2.0.